core/src/main/java/org/elasticsearch/snapshots/SnapshotsService.java

```diff
@@ -991,6 +992,16 @@ public void deleteSnapshot(final Snapshot snapshot, final DeleteSnapshotListener
 
             @Override
             public ClusterState execute(ClusterState currentState) throws Exception {
+                RestoreInProgress restoreInProgress = currentState.custom(RestoreInProgress.TYPE);
+                if (restoreInProgress != null) {
+                    for (RestoreInProgress.Entry entry : restoreInProgress.entries()) {
+                        if (entry.snapshot().equals(snapshot)) {
```

Comment:
Can limiting this to only one snapshot cause some race conditions on repo? The only scenario I can think of is if restore lists index files and just before reading the file it needs delete creates a new generation index file.